src/api/goal/Goals.ts

```diff
@@ -31,3 +34,100 @@ export class Goals {
 export function isGoals(a: any): a is Goals {
     return !!(a as Goals).goals;
 }
+
+/**
+ *  Builder to build Goals instances.
+ */
+export interface GoalsBuilder {
+
+    /**
+     * Plan the given goal or goals to this Goals instance.
+     * @param {Goal | Goals} goals
+     * @returns {Goals & GoalsAndPreConditionBuilder}
+     */
+    plan(...goals: Array<Goal | Goals>): Goals & GoalsAndPreConditionBuilder;
+
+}
+
+/**
+ * Extension to GoalsBuilder allowing to add preConditions,
+ */
+export interface GoalsAndPreConditionBuilder extends GoalsBuilder {
+
+    /**
+     * Add preCondition(s) to previously planned goal or goals.
+     * Note:
+     * This only will effect the goal or goals that where planned immediately
+     * before calling after. Additional a call to this method will remove all pre conditions
+     * that may have existed on the original goal.
+     * @param {Goal} goals
+     * @returns {Goals & GoalsBuilder}
+     */
+    after(...goals: Array<Goal>): Goals & GoalsBuilder;
+
+}
+
+/**
+ * Create Goals instance using a fluent API.
+ *
+ *  const simpleGoals = goals("Simple Goals")
+ *     .plan(ReviewGoal)
+ *     .plan(BuildGoal, AutofixGoal).after(ReviewGoal)
+ *     .plan(StagingEndpointGoal).after(BuildGoal)
+ *     .plan(ProductionDeploymentGoal).after(BuildGoal, StagingEndpointGoal);
+ *
```

Comment:
This could probably use some Markdown love to get formatted properly in Typedoc. We should consider the formatting.